gnmi_server/server.go

```diff
@@ -274,6 +277,26 @@ func (s *Server) checkEncodingAndModel(encoding gnmipb.Encoding, models []*gnmip
 	return nil
 }
 
+func ParseOrigin(origin string, paths []*gnmipb.Path) (string, error) {
+	if len(paths) == 0 {
+		return origin, nil
+	}
+	for i, path := range paths {
+		if origin == "" {
+			if i == 0 {
+				origin = path.Origin
+			}
+		} else if origin != path.Origin {
+			return "", status.Error(codes.Unimplemented, "Origin conflict in path")
+		}
+	}
+	return origin, nil
+}
+
+func IsNativeOrigin(origin string) bool {
+	return origin == "sonic-db" || origin == "sonic-yang"
```

Comment:
>"sonic-db" [](http://example.com/codeflow?start=18&length=10)

Origin `sonic-db` is named as `native` since the gnmi client directly manipulate raw ConfigDB. `sonic-yang` is not considered `native` because there will be a translation between sonic-yang payload to ConfigDB schema.